hybrasyl/CreatureStatus.cs

```diff
@@ -136,39 +151,38 @@ public class CreatureStatus : ICreatureStatus
 
         public double ElapsedSinceTick => (DateTime.Now - LastTick).TotalSeconds;
 
-        private void _init(Status xmlstatus, Creature target, Castable castable)
+        public CreatureStatus(Status xmlstatus, Creature target, Creature source=null, Castable castable = null,
+            int durationOverride = -1, int tickOverride = -1)
         {
-            XMLStatus = xmlstatus;
             Target = target;
+            Source = source;
+            XMLStatus = xmlstatus;
             Castable = castable;
-
-        }
-        public CreatureStatus(Status xmlstatus, Creature target, Castable castable = null,
-            int durationOverride = -1, int tickOverride = -1)
-        {
-            _init(xmlstatus, target, castable);
             Start = DateTime.Now;
             _durationOverride = durationOverride;
             _tickOverride = tickOverride;
         }
 
-        public CreatureStatus(Status xmlstatus, Creature target, Castable castable = null)
+        public CreatureStatus(Status xmlstatus, Creature target, Creature source=null, Castable castable = null)
         {
-            _init(xmlstatus, target, castable);
+            Target = target;
+            Source = source;
+            XMLStatus = xmlstatus;
```

Comment:
What is XMLStatus? Proper property name would be XmlStatus. Even then it should just be 'Status'.

Reply:
This would have been a comment to include in the original review for statuses (note that this is used extensively here and did not appear in this commit).

However, since it does not follow convention, I have fixed it.

`XmlStatus` is a `Statuses.Status` structure.